### PR TITLE
#165413065 Add the user login functionality

### PR DESCRIPTION
**Pull request by Kabalisa**
Description
=======
This feature is about allowing the user to login. users provide their email and password as credentials.

Type of change
=======
N/A

How has it been tested
=======
- pull this branch
- run ```npm install```
- run ```npm start``` to start our react app
-then provide valid credentials to login into our app.

Checklist:
=======
N/A

Pivotal tracker story ID
=======
[#165413065](https://www.pivotaltracker.com/story/show/165413065)

### CODE REVIEW

File: src/components/authentication/login.jsx [hunk 1]
```diff
@@ -1,8 +1,82 @@
...
+      this.props.credentialsValidation(result);
     }
+  };
+
+  componentDidUpdate() {
+    const { token } = this.props.auth;
+    const auth = this.props.auth;
+    const { isAuthanticated } = auth;
+    if (isAuthanticated && token) {
+      if (localStorage.token) {
```
Comment (AJAkimana) [inline src/components/authentication/login.jsx:36]: Try to grab helper @Niyitangasam created and use it to store token

**Review by AJAkimana — COMMENTED** [review]
Good work @Kabalisa I left a comment, you can take a look at it

**Review by Niyitangasam — APPROVED** [review]

**Review by jabichris — DISMISSED** [review]

**Review by jabichris — COMMENTED** [review]
why is your travis failing now?

**Review by jabichris — APPROVED** [review]

File: .travis.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 language: node_js
 
-node_js: '10.15.3'
+node_js: "10.15.3"
```
Comment (macphilips) [inline .travis.yml]: use single-quote over double

File: .travis.yml [hunk 3]
```diff
@@ -22,7 +22,7 @@ env:
   global: CC_TEST_REPORTER_ID=$cc_token
 
 before_install:
-  - '/sbin/start-stop-daemon --start --quiet --pidfile /tmp/custom_xvfb_99.pid --make-pidfile --background --exec /usr/bin/Xvfb -- :99 -ac -screen 0 1280x1024x16'
+  - "/sbin/start-stop-daemon --start --quiet --pidfile /tmp/custom_xvfb_99.pid --make-pidfile --background --exec /usr/bin/Xvfb -- :99 -ac -screen 0 1280x1024x16"
```
Comment (macphilips) [inline .travis.yml]: use single-quote over double

File: cypress/integration/examples/login.spec.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+/* eslint-disable no-undef */
+describe("end-to-end sample test", () => {
+  it("visit google.com", () => {
+    cy.visit("/login");
+    cy.get("#1")
+      .type("ikabalisa20@gmail.com")
+      .should("have.value", "ikabalisa20@gmail.com");
+    cy.get("#2")
```
Comment (macphilips) [inline cypress/integration/examples/login.spec.js]: Why #1 and #2?

File: src/reducers/loginReducer.js [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+            ...state.credentials,
+            isValid: true
+          },
+          isAuthanticated: true,
+          user: { ...User },
+          token: action.payload.data.token
+        };
+      }
+      return state;
+    case LOGIN_ERROR:
```
Comment (macphilips) [inline src/reducers/loginReducer.js]: rejected(LOGIN_ERROR)

File: src/actions/login.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    payload: user
+  };
+};
+
+export const loginUser = ({ email, password }) => dispatch => {
+  dispatch({
+    type: LOGIN_USER,
+    payload: http
+      .post("/api/v1/users/login", { email, password })
+      .catch(error => {
```
Comment (macphilips) [inline src/actions/login.js]: remove this since redux-promise already handles dispatching failed events

**Pull request merged**